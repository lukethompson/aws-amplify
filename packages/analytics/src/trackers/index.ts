export { default as PageViewTracker } from './PageViewTracker';
export { default as EventTracker } from './EventTracker';