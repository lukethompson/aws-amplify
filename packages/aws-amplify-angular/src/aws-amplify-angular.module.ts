import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AmplifyService } from './providers/amplify.service';

import {
  PhotoPickerComponent,
  S3ImageComponent,
  S3AlbumComponent
} from './components/storage';
import {
  AuthenticatorComponentCore,
  RequireNewPasswordComponent,
  GreetingsComponent,
  FormComponent,
  AuthenticatorComponent,
  AuthClass,
  AuthenticatorIonicComponent,

  ConfirmSignInComponent,
  ConfirmSignInComponentCore,
  ConfirmSignInComponentIonic,

  ConfirmSignUpComponent,
  ConfirmSignUpComponentCore,
  ConfirmSignUpComponentIonic,

  SignUpComponent,
  SignUpComponentCore,
  SignUpComponentIonic,

  SignInComponent,
  SignInComponentCore,
  SignInComponentIonic,

  ForgotPasswordComponent,
  ForgotPasswordComponentCore,
  ForgotPasswordComponentIonic

} from './components';

import { DynamicComponentDirective }          from './directives/dynamic.component.directive';


import {IonicModule } from 'ionic-angular';



@NgModule({
  imports: [
    CommonModule,
    IonicModule
  ],
  declarations: [
  	PhotoPickerComponent,
    S3ImageComponent,
    S3AlbumComponent,
    SignInComponentCore,
    AuthenticatorComponentCore,
    RequireNewPasswordComponent,
    SignUpComponent,
    ConfirmSignUpComponent,
    ForgotPasswordComponent,
    GreetingsComponent,
    FormComponent,
    DynamicComponentDirective,
    AuthenticatorComponent,
    AuthenticatorIonicComponent,
    SignInComponentIonic,
    SignUpComponentIonic,
    SignUpComponentCore,
    ConfirmSignUpComponentCore,
    ConfirmSignUpComponentIonic,
    ForgotPasswordComponent,
    ForgotPasswordComponentCore,
    ForgotPasswordComponentIonic,
    ConfirmSignInComponent,
    ConfirmSignInComponentCore,
    ConfirmSignInComponentIonic,
  ],
  entryComponents: [
    AuthenticatorComponentCore,
    AuthenticatorIonicComponent,
    SignInComponentIonic,
    SignUpComponentIonic,
    SignUpComponentCore,
    ConfirmSignUpComponent,
    ConfirmSignUpComponentCore,
    ConfirmSignUpComponentIonic,
    SignUpComponent,
    ForgotPasswordComponent,
    ForgotPasswordComponentCore,
    ForgotPasswordComponentIonic,
    ConfirmSignInComponent,
    ConfirmSignInComponentCore,
    ConfirmSignInComponentIonic,
  ],
  providers: [ AmplifyService ],
  exports: [
  	PhotoPickerComponent,
    S3ImageComponent,
    S3AlbumComponent,
    SignInComponentCore,
    AuthenticatorComponentCore,
    RequireNewPasswordComponent,
    ConfirmSignUpComponent,
    ForgotPasswordComponent,
    GreetingsComponent,
    AuthenticatorComponent,
    DynamicComponentDirective,
    AuthenticatorIonicComponent,
    SignInComponentIonic,
    ConfirmSignUpComponent,
    ConfirmSignUpComponentCore,
    ConfirmSignUpComponentIonic,

    SignUpComponent,
    SignUpComponentIonic,

    ConfirmSignInComponent,
    ConfirmSignInComponentCore,
    ConfirmSignInComponentIonic,

    ForgotPasswordComponent,
    ForgotPasswordComponentCore,
    ForgotPasswordComponentIonic
  ]
})
export class AmplifyAngularModule { }
