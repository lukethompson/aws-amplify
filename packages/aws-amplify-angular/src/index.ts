export { AmplifyAngularModule } from './aws-amplify-angular.module';
export { AmplifyIonicModule } from './aws-amplify-ionic-module';

export { AuthenticatorComponent } from './components/authenticator/authenticator/authenticator.factory';
export { AuthenticatorComponentCore } from './components/authenticator/authenticator/authenticator.component.core';
export { ConfirmSignInComponent } from './components/authenticator/confirm-sign-in-component/confirm-sign-in.factory';
export { ConfirmSignInComponentCore } from './components/authenticator/confirm-sign-in-component/confirm-sign-in-component.core';
export { ConfirmSignUpComponent } from './components/authenticator/confirm-sign-up-component/confirm-sign-up.factory';
export { ConfirmSignUpComponentCore } from './components/authenticator/confirm-sign-up-component/confirm-sign-up.component.core';
export { SignInComponent } from './components/authenticator/sign-in-component/sign-in.component.factory';
export { SignInComponentCore } from './components/authenticator/sign-in-component/sign-in.component.core';
export { SignUpComponent } from './components/authenticator/sign-up-component/sign-up.factory';
export { SignUpComponentCore } from './components/authenticator/sign-up-component/sign-up.component.core';
export { RequireNewPasswordComponent } from './components/authenticator/require-new-password-component/require-new-password.factory';
export { RequireNewPasswordComponentCore } from './components/authenticator/require-new-password-component/require-new-password.component.core';
export { GreetingComponent } from './components/authenticator/greeting-component/greeting.factory';
export { GreetingComponentCore } from './components/authenticator/greeting-component/greeting.component.core';
export { ForgotPasswordComponent } from './components/authenticator/forgot-password-component/forgot-password.factory';
export { ForgotPasswordComponentCore } from './components/authenticator/forgot-password-component/forgot-password.component.core';
export { S3AlbumComponent } from './components/storage/s3-album-component/s3-album.factory';
export { S3AlbumComponentCore } from './components/storage/s3-album-component/s3-album.component.core';
export { S3ImageComponent } from './components/storage/s3-image-component/s3-image.factory';
export { S3ImageComponentCore } from './components/storage/s3-image-component/s3-image.component.core';
export { PhotoPickerComponent } from './components/storage/photo-picker-component/photo-picker.factory';
export { PhotoPickerComponentCore } from './components/storage/photo-picker-component/photo-picker.component.core';
export { ChatBotComponent } from './components/interactions/chatbot/chatbot.factory';
export { ChatbotComponentCore } from './components/interactions/chatbot/chatbot.component.core'; 
export { DynamicComponentDirective } from './directives/dynamic.component.directive';
export { FormComponent } from './components/common/form.component';

export { AuthenticatorIonicComponent } from './ionic/authenticator.component.ionic';
export { ConfirmSignInComponentIonic } from './ionic/confirm-sign-in-component.ionic';
export { ConfirmSignUpComponentIonic } from './ionic/confirm-sign-up.component.ionic';
export { ForgotPasswordComponentIonic } from './ionic/forgot-password.component.ionic';
export { GreetingComponentIonic } from './ionic/greeting.component.ionic';
export { SignInComponentIonic } from './ionic/sign-in.component.ionic';
export { SignUpComponentIonic } from './ionic/sign-up.component.ionic';
export { RequireNewPasswordComponentIonic } from './ionic/require-new-password.component.ionic';
export { PhotoPickerIonicComponent } from './ionic/photo-picker.component.ionic';
export { S3AlbumComponentIonic } from './ionic/s3-album.component.ionic';
export { S3ImageComponentIonic } from './ionic/s3-image.component.ionic';
export { ChatbotComponentIonic } from './ionic/chatbot.component.ionic';