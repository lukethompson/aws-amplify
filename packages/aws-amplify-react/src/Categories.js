import { Amplify } from '@aws-amplify/core';

const Auth = Amplify.Auth;
const API = Amplify.API;
const Analytics = Amplify.Analytics;
const Storage = Amplify.Storage;
const PubSub = Amplify.PubSub;
const Interactions = Amplify.Interactions;

export { Auth, API, Analytics, Storage, PubSub, Interactions };