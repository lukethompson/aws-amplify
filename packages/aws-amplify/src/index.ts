/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import Analytics, { AnalyticsClass, AnalyticsProvider } from './Analytics';
import Auth, { AuthClass } from './Auth';
import Storage, { StorageClass } from './Storage';
import API, { APIClass } from './API';
import I18n from './I18n';
import Cache from './Cache';
import {
    ConsoleLogger as Logger,
    Hub,
    JS,
    ClientDevice,
    Signer
} from './Common';

const logger = new Logger('Amplify');

export default class Amplify {
    static Auth: AuthClass = null;
    static Analytics: AnalyticsClass = null;
    static API: APIClass = null;
    static Storage: StorageClass = null;
    static I18n = null;
    static Cache = null;

    static Logger = null;

    static configure(config) {
        if (!config) { return; }
        Auth.configure(config);
        I18n.configure(config);
        Analytics.configure(config);
        API.configure(config);
        Storage.configure(config);
        Cache.configure(config);

        return config;
    }

    static addPluggable(pluggable) {
        if (pluggable && pluggable['getCategory'] && typeof pluggable['getCategory'] === 'function') {
            const category = pluggable.getCategory();
            switch(category) {
                case 'Analytics':
                    Analytics.addPluggable(pluggable);
                    break;
                case 'Auth':
                    break;
                case 'API':
                    break;
                case 'Cache':
                    break;
                case 'Storage':
                    break;
                default:
                    break;
            }
        }
    }
}

Amplify.Auth = Auth;
Amplify.Analytics = Analytics;
Amplify.API = API;
Amplify.Storage = Storage;
Amplify.I18n = I18n;
Amplify.Cache = Cache;

Amplify.Logger = Logger;

export { Auth, Analytics, Storage, API, I18n, Logger, Hub, Cache, JS, ClientDevice, Signer };
export { AuthClass, AnalyticsClass, APIClass, StorageClass, AnalyticsProvider };
