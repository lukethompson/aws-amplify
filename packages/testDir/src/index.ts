/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import Auth, { AuthClass } from '@aws-amplify/auth';

import Cache from '@aws-amplify/cache';
import {
    ConsoleLogger as Logger,
    Hub,
    JS,
    ClientDevice,
    Signer,
    I18n,
    Amplify,
    ServiceWorker
} from '@aws-amplify/core';

export default Amplify;

Amplify.Auth = Auth;
Amplify.Storage = Storage;
Amplify.I18n = I18n;
Amplify.Cache = Cache;
Amplify.Logger = Logger;
Amplify.ServiceWorker = ServiceWorker;

export { Auth, I18n, Logger, Hub, Cache, JS, ClientDevice, Signer, ServiceWorker };
export { 
    AuthClass,
};
