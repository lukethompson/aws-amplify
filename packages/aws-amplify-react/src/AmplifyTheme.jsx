/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import { Hub, ClientDevice } from 'aws-amplify';

export const Container = {
    fontFamily: `-apple-system,
                BlinkMacSystemFont,
                "Segoe UI",
                Roboto,
                "Helvetica Neue",
                Arial,
                sans-serif,
                "Apple Color Emoji",
                "Segoe UI Emoji",
                "Segoe UI Symbol"`,
    textAlign: 'center'
};

export const Section = {
    fontWeight: '200'
};

export const FormSection = {
    display: 'inline-block',
    fontWeight: '200'
};

export const ErrorSection = {
    background: 'orange',
    padding: '0.5em',
    margin: '0.2em auto',
    fontWeight: '300'
};

export const NavBar = {
    padding: '0.5em 0.5em 0.2em',
    marginBottom: '0.5em',
    borderBottom: '1px solid #00fff6'
};

export const NavRight = {
    textAlign: 'right',
    fontSize: '0.8em'
};

export const NavButton = {
    padding: '0',
    border: 'none',
    borderBottom: '1px solid #007bff',
    margin: 'auto 0.3em',
    background: '#fff',
    color: '#007bff'
}

export const SectionHeader = {
    background: '#007bff',
    color: '#fff',
    padding: '0.6em',
    fontSize: '1.2em',
    fontWeight: 'normal',
    textAlign: 'center'
};

export const SectionBody = {
    background: '#fff',
    color: '#000',
    padding: '1em',
    fontSize: '1em'
};

export const SectionFooter = {
    background: '#fff',
    color: '#000',
    padding: '1em',
    fontSize: '0.8em',
    textAlign: 'left',
    borderBottom: '1px solid #00fff6'
};

export const FormRow = {
    marginBottom: '0.5em',
    textAlign: 'left'
};

export const ActionRow = {
    marginTop: '1.5em',
    textAlign: 'right'
};

export const Input = {
    fontSize: '1em',
    fontWeight: '200',
    width: '15em',
    padding: '4px 2px',
    border: '1px solid #aaa'
};

export const Button = {
    fontSize: '1em',
    padding: '0.3em 0.5em',
    margin: 'auto 0.3em',
    border: '1px solid #007bff',
    background: '#fff',
    color: '#007bff'
}

export const Album = {
    display: 'block',
    width: '100%',
    textAlign: 'left'
}

export const AlbumPhoto = {
    display: 'inline-block',
    width: '32.5%',
    margin: '0.2%',
    verticalAlign: 'top'
}

export const Photo = {
    display: 'inline-block',
    width: '32.5%',
    margin: '0.2%',
    verticalAlign: 'top'
}

export const PhotoImg = {
    maxWidth: '100%'
}

export const AlbumText = {
    display: 'inline-block',
    width: '32.5%',
    margin: '0.2%',
    verticalAlign: 'top',
    maxHeight: 320,
    overflow: 'auto'
}

export const Text = {
    display: 'inline-block',
    width: '32.5%',
    margin: '0.2%',
    verticalAlign: 'top',
    maxHeight: 320,
    overflow: 'auto'
}

export const Picker = {
    width: '100%',
    textAlign: 'center'
}

export const PickerPicker = {
    width: '100%',
    textAlign: 'center'
}

export const PickerButton = {
    background: '#fff',
    border: '1px solid #ccc',
    fontWeight: '200',
    width: '100%'
}

export const PickerInput = {
}

export const A = {
    textDecoration: 'underline'
}

export const Label = {
    margin: 'auto 0.5em'
}

export const Pre = {
    textAlign: 'left',
    margin: '0',
    padding: '1em',
    background: '#eee',
    overflow: 'auto'
}

export const Col2 = {
    display: 'inline-block',
    width: '16.6%'
}

export const Col3 = {
    display: 'inline-block',
    width: '24.5%',
    margin: '0.2%'
}

export const Col4 = {
    display: 'inline-block',
    width: '32%',
    margin: '0.2%'
}

export const Col6 = {
    display: 'inline-block',
    width: '49.5%',
    margin: '0.2%'
}

export const Col8 = {
    display: 'inline-block',
    width: '66%',
    margin: '0.2%'
}

export const Col9 = {
    display: 'inline-block',
    width: '74.5%',
    margin: '0.2%'
}

export const Col10 = {
    display: 'inline-block',
    width: '83.6%',
    margin: '0.2%'
}

export const Col12 = {
    display: 'inline-block',
    width: '99.2%',
    margin: '0.2%'
}

export const HalfHeight = {
    height: 320,
    overflow: 'auto'
}

export const Center = {
    textAlign: 'center'
}

export const Hidden = {
    display: 'none'
}

const AmplifyTheme = {
    container: Container,
    navBar: NavBar,
    navRight: NavRight,
    navButton: NavButton,

    section: Section,
    sectionHeader: SectionHeader,
    sectionBody: SectionBody,
    sectionFooter: SectionFooter,

    formSection: FormSection,
    formRow: FormRow,
    actionRow: ActionRow,
    input: Input,
    button: Button,

    errorSection: ErrorSection,

    photo: Photo,
    photoImg: PhotoImg,
    text: Text,
    album: Album,
    albumPhoto: AlbumPhoto,
    albumText: AlbumText,
    picker: Picker,
    pickerPicker: PickerPicker,
    pickerButton: PickerButton,
    pickerInput: PickerInput,

    a: A,
    label: Label,
    pre: Pre,

    col3: Col3,
    col4: Col4,
    col6: Col6,
    col8: Col8,
    col9: Col9,
    col10: Col10,
    col12: Col12,
    halfHeight: HalfHeight,

    center: Center,
    hidden: Hidden
}

class MediaQuery {
    query() {
        const dim = ClientDevice.dimension();
        const { width, height } = dim;

        if (width < 576) {
            AmplifyTheme.albumPhoto = Object.assign({}, AlbumPhoto, Col6);
            AmplifyTheme.photo = Object.assign({}, Photo, Col12);
            AmplifyTheme.albumText = Object.assign({}, AlbumText, Col12);
        } else if (width < 768) {
            AmplifyTheme.albumPhoto = Object.assign({}, AlbumPhoto, Col4);
            AmplifyTheme.photo = Object.assign({}, Photo, Col6);
            AmplifyTheme.albumText = Object.assign({}, AlbumText, Col8);
        } else if (width < 992) {
            AmplifyTheme.albumPhoto = Object.assign({}, AlbumPhoto, Col3);
            AmplifyTheme.photo = Object.assign({}, Photo, Col6);
            AmplifyTheme.albumText = Object.assign({}, AlbumText, Col6);
        } else {
            AmplifyTheme.albumPhoto = Object.assign({}, AlbumPhoto, Col2);
            AmplifyTheme.photo = Object.assign({}, Photo, Col4);
            AmplifyTheme.albumText = Object.assign({}, AlbumText, Col4);
        }

        AmplifyTheme.halfHeight = Object.assign({}, HalfHeight, { height: height / 2 });
        AmplifyTheme.text = Object.assign({}, AmplifyTheme.text, { maxHeight: height / 2 });
        AmplifyTheme.albumText = Object.assign({}, AmplifyTheme.albumText, { maxHeight: height / 3 });
    }

    onHubCapsule() {
        this.query();
    }
}

const mediaQuery = new MediaQuery();
mediaQuery.query();
Hub.listen('window', mediaQuery, 'AmplifyTheme');

export default AmplifyTheme;
