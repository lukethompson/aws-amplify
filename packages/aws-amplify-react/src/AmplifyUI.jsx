/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import React from 'react';

import { JS } from 'aws-amplify';

import AmplifyTheme from './AmplifyTheme';

export const FormSection = (props) => {
    const theme = props.theme || AmplifyTheme;
    return beforeAfter(
        <div className="amplify-form-section" style={theme.formSection}>
            {props.children}
        </div>
    )
}

export const SectionHeader = (props) => {
    const theme = props.theme || AmplifyTheme;
    return beforeAfter(
        <div className="amplify-section-header" style={theme.sectionHeader}>
            {props.children}
        </div>
    )
}

export const SectionFooter = (props) => {
    const theme = props.theme || AmplifyTheme;
    return beforeAfter(
        <div className="amplify-section-footer" style={theme.sectionFooter}>
            {props.children}
        </div>
    )
}

export const SectionBody = (props) => {
    const theme = props.theme || AmplifyTheme;
    return beforeAfter(
        <div className="amplify-section-body" style={theme.sectionBody}>
            {props.children}
        </div>
    )
}

export const ActionRow = (props) => {
    const theme = props.theme || AmplifyTheme;
    return beforeAfter(
        <div className="amplify-action-row" style={theme.actionRow}>
            {props.children}
        </div>
    )
}

export const FormRow = (props) => {
    const theme = props.theme || AmplifyTheme;
    return beforeAfter(
        <div className="amplify-form-row" style={theme.formRow}>
            {props.children}
        </div>
    )
}

export const InputRow = (props) => {
    const theme = props.theme || AmplifyTheme;
    const p = JS.objectLessAttributes(props, 'theme');
    return (
        <FormRow theme={theme}>
            {beforeAfter(
                <input
                    {...p}
                    className="amplify-input"
                    style={theme.input}
                />
            )}
        </FormRow>
    )
}

export const RadioRow = (props) => {
    const id = props.id || '_' + props.value;
    const theme = props.theme || AmplifyTheme;
    return (
        <FormRow theme={theme}>
            <Radio
                {...props}
                id={id}
            />
            <Label
                htmlFor={id}
                theme={theme}
            >{props.placeholder}</Label>
        </FormRow>
    )
}

export const Radio = (props) => {
    const theme = props.theme || AmplifyTheme;
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
            <input
                {...p}
                type="radio"
                className="amplify-radio"
                style={theme.radio}
            />
    )
}

export const CheckboxRow = (props) => {
    const id = props.id || '_' + props.name;
    const theme = props.theme || AmplifyTheme;
    return (
        <FormRow theme={theme}>
            <Checkbox
                {...props}
                id={id}
            />
            <Label
                htmlFor={id}
                theme={theme}
            >{props.placeholder}</Label>
        </FormRow>
    )
}

export const Checkbox = (props) => {
    const theme = props.theme || AmplifyTheme;
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
        <input
            {...p}
            type="checkbox"
            className="amplify-checkbox"
            style={theme.checkbox}
        />
    )
}

export const MessageRow = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <FormRow theme={theme}>
            <MessageContent theme={theme}>{props.children}</MessageContent>
        </FormRow>
    )
}

export const MessageContent = (props) => {
    const theme = props.theme || AmplifyTheme;
    return beforeAfter(
        <span className="amplify-message-content" style={theme.messageContent}>
            {props.children}
        </span>
    )
}

export const ButtonRow = (props) => {
    const theme = props.theme || AmplifyTheme;
    return beforeAfter(
        <div className="amplify-action-row" style={theme.actionRow}>
            <Button {...props} />
        </div>
    )
}

export const Button = (props) => {
    const theme = props.theme || AmplifyTheme;
    const p = JS.objectLessAttributes(props, 'theme');

    return beforeAfter(
        <button {...p} className="amplify-button" style={theme.button}>
            <ButtonContent theme={theme}>{props.children}</ButtonContent>
        </button>
    )
}

export const ButtonContent = (props) => {
    const theme = props.theme || AmplifyTheme;
    return beforeAfter(
        <span className="amplify-button-content" style={theme.buttonContent}>
            {props.children}
        </span>
    )
}

export const NavButton = (props) => {
    const theme = props.theme || AmplifyTheme;
    const p = JS.objectLessAttributes(props, 'theme');

    return beforeAfter(
        <button {...p} className="amplify-nav-button" style={theme.navButton}>
            {beforeAfter(
                <span style={theme.navButtonContent}>{props.children}</span>
            )}
        </button>
    )
}

export const Link = (props) => {
    const theme = props.theme || AmplifyTheme;
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
        <a {...p} className="amplify-a" style={theme.a}>{props.children}</a>
    )
}

export const Label = (props) => {
    const theme = props.theme || AmplifyTheme;
    const p = JS.objectLessAttributes(props, 'theme');
    return beforeAfter(
        <label {...p} className="amplify-label" style={theme.label}>{props.children}</label>
    )
}

export const NavBar = (props) => {
    const theme = props.theme || AmplifyTheme;
    return beforeAfter(
        <div className="amplify-nav-bar" style={theme.navBar}>
            {props.children}
        </div>
    )
}

export const NavRight = (props) => {
    const theme = props.theme || AmplifyTheme;
    return beforeAfter(
        <div className="amplify-nav-right" style={theme.navRight}>
            {props.children}
        </div>
    )
}

export const beforeAfter = (el) => {
    const style = el.props.style || {};
    const { before, after } = style;
    if (!before && !after) { return el; }

    return (
        <span>
            {before? <span style={before}>{before.content}</span> : null}
            {el}
            {after? <span style={after}>{after.content}</span> : null}
        </span>
    )
}

export const transparent1X1 = 'data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==';

export const white1X1 = 'data:image/gif;base64,R0lGODlhAQABAIAAAP7//wAAACH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==';
