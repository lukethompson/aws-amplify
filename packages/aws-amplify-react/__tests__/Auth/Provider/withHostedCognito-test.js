import React, { Component } from 'react';
import withHostedCognito, { HostedCognitoButton } from '../../../src/Auth/Provider/withHostedCognito';
import { SignInButton, Button } from '../../../src/AmplifyUI';
import { Auth } from 'aws-amplify';

describe('withHostedCognito test', () => {
    describe('render test', () => {
        test('render correctly', () => {
            const MockComp = class extends Component {
                render() {
                    return <div />;
                }
            }

            const Comp = withHostedCognito(MockComp);
            const wrapper = shallow(<Comp/>);
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('signIn test', () => {
        test('happy case with connected response', () => {
            const MockComp = class extends Component {
                render() {
                    return <div />;
                }
            }

            const spyon = jest.spyOn(Auth, 'configure').mockImplementationOnce(() => {
                return {
                    hostedUIOptions: {
                        AppWebDomain: 'domain',
                        RedirectUriSignIn: 'redirectUriSignIn',
                        RedirectUriSignOut: 'redirectUriSignOut',
                        ResponseType: 'responseType'
                    },
                    userPoolWebClientId: 'userPoolWebClientId'
                }
            })
            const Comp = withHostedCognito(MockComp);
            const wrapper = shallow(<Comp/>);
            const comp = wrapper.instance();

            comp.signIn();

            spyon.mockClear();
        });
    });
});