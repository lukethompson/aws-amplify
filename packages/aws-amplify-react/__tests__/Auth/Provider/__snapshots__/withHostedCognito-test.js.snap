// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withHostedCognito test render test render correctly 1`] = `
<MockComp
  hostedCognitoSignIn={[Function]}
/>
`;

exports[`withHostedCognito test render test render correctly with hosted cognito button 1`] = `
<Button
  hostedCognitoSignIn={[Function]}
/>
`;
